plugins/builds/artifacts/get.js

```diff
@@ -61,7 +63,24 @@ module.exports = config => ({
                         baseUrl += `&type=${request.query.type}`;
                     }
         
-                    return h.redirect(baseUrl);
+                    const gotStream = got.stream(baseUrl);
+                    let response = h.response(gotStream);
+
+                    if (request.query.type === 'download') {
+                        response.headers['content-type'] = 'application/octet-stream';
+                        response.headers['content-disposition'] = `attachment; filename="${encodeURI(encodedArtifact)}"`;
+                    } else if (request.query.type === 'preview') {
+                        const fileExt = encodedArtifact.split('.').pop();
+                        const mime = getMimeFromFileName(fileExt, encodedArtifact);
+
+                        response.headers['content-type'] = mime;
```

Comment:
why are we computing these again at API ? Store is already doing this and setting appropriate headers. API  should be reading these data from store webservice response headers.